core/src/main/java/org/frankframework/http/authentication/SamlAssertionOauth.java

```diff
@@ -0,0 +1,259 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.http.authentication;
+
+import java.io.ByteArrayInputStream;
+import java.io.StringWriter;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.List;
+import java.util.UUID;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.auth.Credentials;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.xml.security.algorithms.MessageDigestAlgorithm;
+import org.apache.xml.security.signature.XMLSignature;
+import org.apache.xml.security.transforms.Transforms;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.encryption.EncryptionException;
+import org.frankframework.encryption.PkiUtil;
+import org.frankframework.http.AbstractHttpSession;
+
+public class SamlAssertionOauth extends AbstractOauthAuthenticator {
+
+	private static final String SAML2_BEARER_GRANT_TYPE = "urn:ietf:params:oauth:grant-type:saml2-bearer";
+
+	private PrivateKey privateKey;
+	private X509Certificate certificate;
+
+	static {
+		org.apache.xml.security.Init.init();
+	}
+
+	public SamlAssertionOauth(AbstractHttpSession session) throws HttpAuthenticationException {
+		super(session);
+	}
+
+	@Override
+	public void configure() throws ConfigurationException {
+		if (session.getClientId() == null) {
+			throw new ConfigurationException("clientId is required");
+		}
+
+		if (session.getClientSecret() == null) {
+			throw new ConfigurationException("clientSecret is required");
+		}
+
+		try {
+			privateKey = PkiUtil.getPrivateKey(session, "Creation of SAML assertion");
+		} catch (EncryptionException e) {
+			throw new ConfigurationException(e);
+		}
+
+		try {
```

Comment:
De catch is twee keer hetzelfde (`catch (EncryptionException e)`). Kan je de twee try blokken combineren? Leest wat lekkerder weg ;-)